pca_for_index/pca_analysis.Rmd

```diff
@@ -0,0 +1,439 @@
+
+
+hist(hq_index, breaks=100)
+summary(hq_index)
+
+length(which(hq_index>5))
+```
+
+
+We standardize the index (SI) to a range from 0 to 10 (lowest to highest housing quality), as follows:
+
+$SI_j = \frac{NSCI_j-Min(NSCI)}{Max(NSCI)-Min(NSCI)} * 10$
+
+```{r}
+std_hq_index <- (hq_index-min(hq_index)) / (max(hq_index)-min(hq_index)) * 10
+summary(std_hq_index)
+
+hist(std_hq_index, breaks=100)
+```
+
+
+```{r}
+# Create dataset that contains component scores and standardized index
+df_index <- scores %>% mutate(housing_qual_index = std_hq_index)
+```
+
+# Alternative way to build the index
+```{r}
+# Compute variable loadings 
+var_loadings <- sweep(result_pca$var$coord,2,sqrt(result_pca$eig[1:ncol(result_pca$var$coord),1]),FUN="/")
+var_loadings <- data.frame(var_loadings)
+var_loadings$var <- rownames(var_loadings)
+rownames(var_loadings) <- c(1:nrow(var_loadings))
+var_loadings <- var_loadings %>% select(var, Dim.1, Dim.2, Dim.3, Dim.4, Dim.5, Dim.6)
+var_loadings
+
+# Find component "scores" for each individual based on the variable loadings and the individuals' variable value 
+indv_var_values <- as.matrix(df_quality_vars_lumped_reduced_num) # individuals' variable value
+comp1_score <- indv_var_values %*% var_loadings$Dim.1
+comp2_score <- indv_var_values %*% var_loadings$Dim.2
+comp3_score <- indv_var_values %*% var_loadings$Dim.3
+comp4_score <- indv_var_values %*% var_loadings$Dim.4
+comp5_score <- indv_var_values %*% var_loadings$Dim.5
+comp6_score <- indv_var_values %*% var_loadings$Dim.6
+
+
+# Convert scores to binary variables according to threshold (75% quartile)
+comp1_score_binary <- ifelse(comp1_score > quantile(comp1_score, c(0.75), na.rm = TRUE), 1, 0)
+comp2_score_binary <- ifelse(comp2_score > quantile(comp2_score, c(0.75), na.rm = TRUE), 1, 0)
+comp3_score_binary <- ifelse(comp3_score > quantile(comp3_score, c(0.75), na.rm = TRUE), 1, 0)
+comp4_score_binary <- ifelse(comp4_score > quantile(comp4_score, c(0.75), na.rm = TRUE), 1, 0)
+comp5_score_binary <- ifelse(comp5_score > quantile(comp5_score, c(0.75), na.rm = TRUE), 1, 0)
+comp6_score_binary <- ifelse(comp6_score > quantile(comp6_score, c(0.75), na.rm = TRUE), 1, 0)
+
+# Create index as the sum of the binary scores. The index should have a range 0 - 6. However, similar to what was seen in the first housing quality index that was created, only 0.04036582% of the data has scores above 4. Additionally, this method of creating the index does not account for missing values. A total of 3089 observations have missing values and, hence, are not assigned an index.
+
+hq_index2 <- comp1_score_binary + comp2_score_binary + comp3_score_binary + comp4_score_binary + comp5_score_binary + comp6_score_binary
+
+summary(hq_index2)
+length(which(hq_index2 > 4))*100/nrow(df_quality_vars_lumped_reduced_num) # only 0.04% of the data has scores above 4
+hist(hq_index2)
```

Comment:
If we include this plot in the paper or presentation, can we add breaks=5 to the histogram so it makes the x-axis look discrete? Right now we have some space in between each bars, which makes the index look continuous.